### PR TITLE
Migrate all rule tests to resolveFile

### LINKED ISSUE #154: Refactor existing tests to resolveFile approach
Migrate current approach with strings in tests to separate dart files with examples. It will improve maintainability and readability of our tests.

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain: refactor all rule tests and migrate then to `resolveFile`

### Is there anything you'd like reviewers to focus on?
Take a look at RuleTestHelper, seems like it's a good abstraction for rules testing.

Closes #154

### CODE REVIEW

File: test/helpers/rule_test_helper.dart [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  static void verifyInitialization({
+    required Iterable<Issue> issues,
+    required String ruleId,
+    required Severity severity,
+  }) {
+    expect(issues.every((issue) => issue.ruleId == ruleId), isTrue);
+    expect(issues.every((issue) => issue.severity == severity), isTrue);
+  }
+
+  static void verifyIssues({
```
Comment (dkrutskikh) [inline test/helpers/rule_test_helper.dart:36]: @incendial please add ignore `Long method` and `Long Parameter List`

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: test/helpers/rule_test_helper.dart (hunk 1 above)
Comment (incendial) [reply test/helpers/rule_test_helper.dart:36]: Why? No report from the analyzer whatsoever.

File: test/obsoleted/rules/no_equal_arguments/no_equal_arguments_test.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
+@TestOn('vm')
+import 'package:dart_code_metrics/src/models/severity.dart';
+import 'package:dart_code_metrics/src/obsoleted/rules/no_equal_arguments.dart';
+import 'package:test/test.dart';
+
+import '../../../helpers/rule_test_helper.dart';
+
+// ignore_for_file: no_adjacent_strings_in_list
```
[the remark this answers is not in the document]
Comment (incendial) [reply test/obsoleted/rules/no_equal_arguments/no_equal_arguments_test.dart]: Fixed.

File: test/obsoleted/rules/prefer_intl_name/examples/incorrect_example.dart [hunk 3]
```diff
@@ -0,0 +1,242 @@
+import 'package:intl/intl.dart';
```
[the remark this answers is not in the document]
Comment (incendial) [reply test/obsoleted/rules/prefer_intl_name/examples/incorrect_example.dart:1]: Fixed.

File: test/obsoleted/rules/provide_correct_intl_args/examples/incorrect_example.dart [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import 'package:intl/intl.dart';
```
[the remark this answers is not in the document]
Comment (incendial) [reply test/obsoleted/rules/provide_correct_intl_args/examples/incorrect_example.dart:1]: Fixed.

File: test/obsoleted/rules/no_magic_number/examples/exceptions_example.dart [hunk 5]
```diff
@@ -0,0 +1,8 @@
+int good_f1(int x) => x + 1;
+bool good_f2(int x) => x != 0;
+bool good_f3(String x) => x.indexOf(str) != -1;
+final someDay = DateTime(2006, 12, 1);
```
[the remark this answers is not in the document]
Comment (incendial) [reply test/obsoleted/rules/no_magic_number/examples/exceptions_example.dart:4]: Done.

File: lib/src/obsoleted/rules/no_equal_arguments.dart [hunk 6]
```diff
@@ -61,6 +61,13 @@ class _Visitor extends RecursiveAstVisitor<void> {
     _visitArguments(node.argumentList.arguments);
   }
 
+  @override
+  void visitInstanceCreationExpression(InstanceCreationExpression node) {
```
[the remark this answers is not in the document]
Comment (incendial) [reply lib/src/obsoleted/rules/no_equal_arguments.dart:65]: Done. Also added `no_magic_number`.

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,6 +6,7 @@
 * Add static code diagnostic `avoid-returning-widgets`.
 * Improve static code diagnostic `no-boolean-literal-compare`.
 * Add `Source lines of Code` metric.
+* Migrate all rule tests to `resolveFile`.
```
[the remark this answers is not in the document]
Comment (incendial) [reply CHANGELOG.md]: Done.

**Pull request merged**